instance_counter.rb

```diff
@@ -0,0 +1,26 @@
+module InstanceCounter
+
+  def self.included(base)
+    base.extend ClassMethods
+    base.send :include, InstanceMethods
+  end
+
+  module ClassMethods
+
+    def instances
+      @instances ||= 0
+    end 
+
+    def intrans_add
+      @instances = self.instances + 1 #ничего лучше не придумал
```

Comment:
в общем решение верное
если добавить переменной сеттер, то можно так записать:

```suggestion
      self.instances += 1
```